### ISSUE TITLE
Regression: false positive for `Lint/SuppressedException` in blocks

### ISSUE THREAD

**Issue by timoschilling** [rubocop-hq/rubocop#8081]
## Regresion

0.85.0 introduced a regression in `Lint/SuppressedException` in blocks, 0.84.0 didn't find any offences in the example below.

## Expected behavior

No offense should be detected.

## Actual behavior

```
test.rb:4:1: W: Lint/SuppressedException: Do not suppress exceptions.
rescue Foo
^^^^^^^^^^
```

## Steps to reproduce the problem

1. Create a file
```ruby
# frozen_string_literal: true

foo do
rescue Foo
  # ok
end

```

2. Run rubocop on this file

## RuboCop version

```
$ [bundle exec] rubocop -V
0.85.0 (using Parser 2.7.1.3, rubocop-ast 0.0.3, running on ruby 2.6.6 x86_64-linux)
```
